server/services/notify/plugindelivery/plugin_delivery.go

```diff
@@ -91,3 +104,31 @@ func (pd *PluginDelivery) getTeamID(evt notify.BlockChangeEvent) (string, error)
 	}
 	return channel.TeamId, nil
 }
+
+type cardEmbed struct {
+	OriginalPath string `json:"originalPath"`
+	WorkspaceID  string `json:"workspaceID"`
+	BoardID      string `json:"boardID"`
+	CardID       string `json:"cardID"`
+}
+
+func embedLinkInPost(post *model.Post, embed cardEmbed) (*model.Post, error) {
```

Comment:
Are we okay with having this function duplicated?

Reply:
I tried moving this to server model but it created dependencies between plugin and server that shouldn't exist.  It's really a trade-off between the parsing and duplicating the code a bit.